### PR TITLE
binutils: Remove gold linker

### PR DESCRIPTION
**Pull request by orgads**
It doesn't support MinGW, and is generally useless on Windows.

This reverts commit fbb2df5f81d3d983e006ffb97b56f34d6a890d60.

### CODE REVIEW

**Comment by lazka** [comment]
@Alexpux any thoughts on this? Since you added that change.

File: mingw-w64-binutils/PKGBUILD [hunk 1]
```diff
@@ -138,7 +126,6 @@ package() {
   #install -m644 ${srcdir}/${_realname}-${pkgver}/include/libiberty.h ${pkgdir}${MINGW_PREFIX}/include
   #install -m644 ${srcdir}/${_realname}-${pkgver}/include/demangle.h ${pkgdir}${MINGW_PREFIX}/include
 
-  find ${pkgdir}${MINGW_PREFIX}/share -type f -iname "opcodes.mo" -o -iname "bfd.mo" | xargs -rtl1 rm
```
Comment (lazka) [inline mingw-w64-binutils/PKGBUILD:141]: looks like this results in file conflicts with gdb?

File: mingw-w64-binutils/PKGBUILD [hunk 2]
```diff
@@ -108,10 +99,7 @@ build() {
     --enable-nls \
     --disable-rpath \
     --disable-multilib \
-    --enable-install-libiberty \
-    --enable-plugins \
-    --enable-gold \
```
Comment (lazka) [inline mingw-w64-binutils/PKGBUILD:113]: What's with all the other changes?

File: mingw-w64-binutils/0110-binutils-mingw-gnu-print.patch [hunk 3]
```diff
@@ -110,20 +110,6 @@ diff -urN binutils-2.27.orig/binutils/strings.c binutils-2.27/binutils/strings.c
  # else
  		printf ("%7I64x ", (unsigned long long) start);
 diff -urN binutils-2.27.orig/gas/as.h binutils-2.27/gas/as.h
---- binutils-2.27.orig/gas/as.h	2016-08-03 09:36:51.000000000 +0200
```
Comment (lazka) [inline mingw-w64-binutils/0110-binutils-mingw-gnu-print.patch:113]: This doesn't seem gold related

File: mingw-w64-binutils/0002-check-for-unusual-file-harder.patch [hunk 4]
```diff
@@ -1,78 +0,0 @@
-Configure scripts tend to do things like this:
```
Comment (lazka) [inline mingw-w64-binutils/0002-check-for-unusual-file-harder.patch:1]: Is this gold related?

File: mingw-w64-binutils/PKGBUILD (hunk 2 above)
Comment (orgads) [reply mingw-w64-binutils/PKGBUILD:113]: They were all part of the same commit. Are they unrelated? I thought enable-plugins was needed for enable-gold.

Comment (lazka) [reply mingw-w64-binutils/PKGBUILD:113]: I don't know. I think it's clear now that the commit you reverted wasn't split up as much as it would have been :)

Comment (orgads) [reply mingw-w64-binutils/PKGBUILD:113]: Ok, I'll minimize this patch.

**Comment by orgads** [comment]
Better now?

**Comment by lazka** [comment]
Yes, thanks. What does `enable-plugins` change?

Let's wait for @Alexpux to respond

**Comment by orgads** [comment]
Reverted the `enable-plugins` too.

**Comment by lazka** [comment]
Thanks!

**Comment by lazka** [comment]
@mati865 ok to remove gold?

**Comment by mati865** [comment]
AFAICT gold will never support non-ELF targets so there is no point in keeping it. We should wait for Alexey though.

**Comment by lazka** [comment]
I'll merge this in two days if there is no response. I can't find anything re gold + Windows.
